jetty9-base/src/main/java/com/google/cloud/runtimes/jetty9/TracingLogHandler.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.runtimes.jetty9;
+
+import java.util.logging.ConsoleHandler;
+import java.util.logging.LogRecord;
+
+/**
+ * A Logging Handler extended with a request traceId label.
+ */
+public class TracingLogHandler extends ConsoleHandler {
+
+  /**
+   * Construct a TracingLogHandler.
+   */
+  public TracingLogHandler() {}
+
+  @Override
+  public synchronized void publish(LogRecord record) {
+    String traceId = RequestContextScope.getCurrentTraceId();
+    if (traceId != null) {
+      record.setMessage(record.getMessage() + " traceId=" + traceId);
```

Comment:
Problem using parameters is that there is a non programatic dependency between the handler that adds the traceId as an extra parameter and the formatter that has to expect it to be there even if not referenced in the message.

It is also a step away from the direction we wish to eventually take where we will use stackdriver as the log sink and it has a mechanism for transporting the traceid which is not part of the message.